recipes/mpark_variant/all/conanfile.py

```diff
@@ -0,0 +1,54 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+import shutil
+
+
+class VariantConan(ConanFile):
+    name = "mpark_variant"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/mpark/variant"
+    description = "C++17 std::variant for C++11/14/17"
+    license = "BSL-1.0"
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake"
+
+    settings = "os", "arch", "compiler", "build_type"
+
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+    _cmake = None
+
+    def configure(self):
+        tools.check_min_cppstd(self, "11")
```

Comment:
```suggestion
        if self.settings.get_safe("cppstd"):
            tools.check_min_cppstd(self, "11")
```

Reply:
[tools.check_min_cppstd seems to be already using get_safe](https://github.com/conan-io/conan/blob/a1086d3df17b482a8ab6e0ce506251dfc7327381/conans/client/tools/settings.py#L41). Why is it needed to check it here ?

Reply:
Yes, because if cppstd is declared, so self.settings.compiler.cppstd will be used to compare. Otherwise, it will get the default c++ standard supported by the compiler and compare. However, most of compilers supports c++11, but non of them use c++11 by default e.g gcc 4.9